corehq/apps/dump_reload/management/commands/dump_case_data.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+from datetime import datetime
+from django.db import connections
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+from corehq.apps.dump_reload.const import DATETIME_FORMAT
+from corehq.form_processor.models import CommCareCaseSQL
+
+
+class Command(BaseCommand):
+    help = "Dump all case data, per case type per shard in csv files"
+
+    def add_arguments(self, parser):
+        parser.add_argument('--domain')
+        parser.add_argument('--case_type')
+        parser.add_argument('--db_conn', choices=settings.DATABASES)
+
+    def ensure_params(self, for_domain, for_case_type, for_db_conn):
+        if for_case_type:
+            available_case_types = (CommCareCaseSQL.objects.using(for_db_conn).
+                                    values_list('type', flat=True).distinct())
+            if for_case_type not in available_case_types:
+                raise CommandError("Unexpected case type {for_case_type} passed. "
+                                   "Should be from the following: {case_types}"
+                                   .format(for_case_type=for_case_type, case_types=available_case_types))
+
+        if for_domain:
+            available_domains = (CommCareCaseSQL.objects.using(for_db_conn).
+                                 values_list('domain', flat=True).distinct())
+            if for_domain not in available_domains:
+                raise CommandError("Domain name {for_domain} not found for any case in db {for_db_conn}".format(
+                    for_domain=for_domain, for_db_conn=for_db_conn
+                ))
+
+    def handle(self, *args, **options):
+        for_domain = options.get('domain')
+        for_case_type = options.get('case_type')
+        for_db_conn = options.get('db_conn')
+
+        if for_db_conn:
+            self.ensure_params(for_domain, for_case_type, for_db_conn)
+            dbs = [for_db_conn]
+        else:
+            dbs = settings.DATABASES
```

Comment:
how come you don't call `ensure_params` here too?

Reply:
Aha.. right .. i had checks here earlier.. moved that to the method.. should have added that method call for this as well..

Reply:
the check for case type was already present.. added check for domain

`ensure_params` is not called here because if we have a specific db_conn passed we need to just check it for that one table,
else
we check if for each partition/db as we iterate.